packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2851,6 +2842,43 @@ export class ContainerRuntime
 		}
 	}
 
+	private scheduleFlush() {
+		if (this.flushTaskExists) {
+			return;
+		}
+
+		this.flushTaskExists = true;
+		const flush = () => {
+			this.flushTaskExists = false;
+			this.flush();
+		};
+
+		switch (this.flushMode) {
+			case FlushMode.TurnBased:
+				Promise.resolve()
+					.then(() => flush())
+					.catch((error) => {
+						this.closeFn(error as GenericError);
+					});
+				break;
+			case FlushMode.Async:
+				setTimeout(() => {
```

Comment:
We should probably add a feature flag or option that can disable this and use TurnBased instead. It can be used to disable this experiment if something goes wrong because IMO, this is risky can have unintended consequences. Having a kill switch will be helpful.

Reply:
I am thinking a container would try this out by setting flush mode to Async. But if something goes wrong, the only way to fix it would be to revert the change, release a patch and apps will have to consume them.
Having a feature flag short circuits this.

Reply:
Another option is to have a feature flag that can enable this. Then experiments would be to enable it via feature flag (or local storage) and turn it off after experiment is over.

Reply:
Hmmm that makes sense. I was thinking since its our experiment and we can't guarantee that this will work maybe we should provide a way to get out of problematic state. Once this option is provided it cannot be changed right, so documents created with this option will be stuck in this state.

Reply:
would they? I don't think the flushmode is persisted anywhere, right?

Reply:
Oh okay. I misunderstood it then. Flushmode is based on the container runtime options on every create / load :).